borglab/gtsfm
# Restore working rtf viz tool

The packages and result_metrics/ dir had changed since the last time the frontend React JS website compiled. This PR gets the frontend to a compiling state. (Note: package-lock.json is auto-generated and does not need to be reviewed.)

## Testing
0. Install simplejson with `pip install simplejson`.
1. Run the gtsfm pipeline per the README: `python gtsfm/runner/run_scene_optimizer_olssonloader.py --dataset_root tests/data/set1_lund_door --image_extension JPG --config_name deep_front_end.yaml --num_workers 4`
2. Install and run the rtf tool per [rtf_vis_tool/README.md](https://github.com/borglab/gtsfm/pull/524/files#diff-29321d04f70019d4c6f7ca8ef25583de9ff1ff080c557c623600b0967530df1eR5). Simplified instructions are:
```
cd rtf_vis_tool/
npm install --legacy-peer-deps
npm start
(follow the link it generates)
```
3. Click around the UI that appears.

Installation note: if installing node for the first time, make sure to use the LTS version. If a different version of node is installed, you can use [nvm](https://github.com/nvm-sh/nvm) to switch to LTS:
```
nvm install --lts
nvm use --lts
```
